src/main/java/server/ServerRunner.java

```diff
@@ -27,12 +26,18 @@ public void run() {
         try {
             String input = client.chunkStream();
 
-            if(input != null) {
+            if(!input.isEmpty()) {
+                System.out.println("Client request received on "+ Thread.currentThread().getName());
+
                 Request request = requestParser.create(input);
                 Response response = router.generateResponse(request);
                 ResponseFormatter formatter = new ResponseFormatter(response);
 
                 out.print(formatter.stringifyResponse());
+
+                System.out.println("Response sent\n");
```

Comment:
maybe don't need the trailing `\n` if you're using `println`

Reply:
That was just to create some spacing in the console between request/response cycles so it wasn't one long block of text.